src/app/modules/test-navigator/test-navigator.component.ts

```diff
@@ -169,27 +181,119 @@ export class TestNavigatorComponent implements OnInit, OnDestroy {
     await this.updateModel();
   }
 
+  private open(node: TestNavigatorTreeNode) {
+    this.openFilesSubscriptions.remove(node);
+    this.openFilesSubscriptions.add(node, this.messagingService.subscribe(EDITOR_DIRTY_CHANGED, (payload) => {
+      if (node.id === payload.path) {
+        node.dirty = payload.dirty;
+      }
+    }));
+
+    this.messagingService.publish(NAVIGATION_OPEN, node);
+    this.log('published NAVIGATION_OPEN', node);
+  }
+
+
   /** create a new element within the tree */
-  newElement(type: string): void {
-    // TODO: put TREE_NODE_CREATE_NEW on the bus to initiate creation of new file/folder
-    // make sure the (name) validator heeds relevant filter settings
-    // add this new element to the tree
-    // create this new element on the backend (if it fails, remove element from the tree again)
-    // if type is a file, open it right away (issue a NAVIGATION_OPEN (or something alike) command)
+  newElement(type: ElementType): void {
+    const contextNode = this.selectedNode;
+    const parentNode = contextNode.type === ElementType.Folder ? contextNode : contextNode.parent;
+    const payload: TreeViewerInputBoxConfig = {
+      indent: contextNode.type === ElementType.Folder,
+      validateName: (newName: string) => this.validateName(newName, type),
+      onConfirm: async (newName: string) => {
+        const newPath = contextNode.getDirectory() + newName;
+        const requestSuccessful = await this.sendCreateRequest(newPath, type);
+        if (requestSuccessful) {
+          /* const newNode = */ parentNode.addChild({children: [], name: newName, path: newPath, type: type });
+          parentNode.expanded = true;
+          // TODO programmatically select newNode
+          if (type === ElementType.File) {
+            this.messagingService.publish(NAVIGATION_OPEN, { name: newName, id: newPath });
+          }
+        }
+        return requestSuccessful;
+      },
+      iconCssClasses: type === ElementType.Folder ? 'fa-folder' : 'fa-file'
+    };
+    this.messagingService.publish(TREE_NODE_CREATE_AT_SELECTED, payload);
+  }
+
+  renameElement(): void {
+    const selectedNode = this.selectedNode;
+    const payload: InputBoxConfig = {
+      validateName: (newName: string) => this.selectedNode.dirty ?
+        { valid: false, message: 'cannot rename dirty files' } : this.validateName(newName, selectedNode.type),
```

Comment:
should renaming only allow file names (without additional path elements)?
validate name allows for paths, doesn't it?

Reply:
Right. I'll change this so that for both new elements and renaming, only regular names will be allowed.